src/frontend/screens/TrackEdit/index.tsx

```diff
@@ -0,0 +1,101 @@
+import React, {useEffect, useCallback} from 'react';
+import {defineMessages, useIntl} from 'react-intl';
+import {useFocusEffect} from '@react-navigation/native';
+import {Editor} from '../../sharedComponents/Editor';
+import {TrackDescriptionField} from '../SaveTrack/TrackDescriptionField';
+import {usePersistedTrack} from '../../hooks/persistedState/usePersistedTrack';
+import {NativeNavigationComponent} from '../../sharedTypes/navigation';
+import {HeaderLeft} from './HeaderLeft';
+import {SaveButton} from '../../sharedComponents/SaveButton';
+import TrackIcon from '../../images/Track.svg';
+import {useTrackQuery, useEditTrackMutation} from '../../hooks/server/track';
+
+export const m = defineMessages({
+  trackEditScreenTitle: {
+    id: 'screens.TrackEdit.title',
+    defaultMessage: 'Edit Track',
+    description: 'Title for editing track screen',
+  },
+  presetTitle: {
+    id: 'screens.TrackEdit.track',
+    defaultMessage: 'Track',
+    description: 'Preset title for new track screen',
+  },
+});
+
+export const TrackEdit: NativeNavigationComponent<'TrackEdit'> = ({
+  navigation,
+  route,
+}) => {
+  const {formatMessage} = useIntl();
+  useEffect(() => {
+    if (!route.params || !route.params.trackId) {
+      navigation.goBack();
+    }
+  }, [route, navigation]);
+  const {trackId} = route.params as {trackId: string};
```

Comment:
related to other comment about the type for this screen's route param: wondering what situation this is accounting for?

Reply:
See above.

Reply:
I removed the possibility it could be undefined.